rewrite-yaml/src/test/java/org/openrewrite/yaml/MergeYamlTest.java

```diff
@@ -841,4 +841,51 @@ void mergeSequenceMapChangeComplexMapping() {
         );
     }
 
+    @Test
+    void incomingCollapsedKey() {
+        rewriteRun(
+          spec -> spec.recipe(new MergeYaml(
+            "$",
+            //language=yaml
+            "management.endpoints.jmx.unique-names: true",
+            true,
+            null
+          )),
+          yaml(
+            """
+              management:
+                context-path: /admin
+              """, """
+              management:
+                context-path: /admin
+                endpoints.jmx.unique-names: true
+              """
+          )
+        );
+    }
+
+    @Disabled("fix not implemented yet, but potentially an issue")
```

Comment:
This one is definitely an issue to resolve. SnakeYAML will expect for the subkey `endpoints.jmx` not to be duplicated within it's subkeys, but to instead for that subpath to be broken out as described by this test.

Reply:
@joanvr In case you don't have time for this today, can you create an issue? Then I will look into it.

Reply:
I think we may also need another test case to assist with breaking an existing flatten property on the longest common subpath. 

```
management.context-path: /admin
```

Which would then produce the same output as the first test case contained in this PR.